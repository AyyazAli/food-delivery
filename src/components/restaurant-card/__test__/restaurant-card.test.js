import { cleanup, render } from '@testing-library/react';
import React from 'react'
import ReactDOM from 'react-dom'
import { BrowserRouter as Router } from 'react-router-dom';
import RestaurantCard from '../restaurant-card';
import { createBrowserHistory } from 'history';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer'

const history = createBrowserHistory();

afterEach(cleanup)
it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
        <Router>
            <RestaurantCard to="/" />
        </Router>, div)
})


it("renders restaurant card correctly", () => {
    const { getByTestId } = render(
        <Router history={history}>
            <RestaurantCard
                to="/"
                name="KFC"
                description="Best Burgers in town"
                mealType="fastFood"
                action={() => { }} />
        </Router>
    )

    expect(getByTestId('restaurant-card')).toHaveTextContent("KFC")
})

it("matches snapshot", () => {
   const tree = renderer.create(
        <Router history={history}>
            <RestaurantCard
                to="/"
                name="KFC"
                description="Best Burgers in town"
                mealType="fastFood"
                action={() => { }} />
        </Router>
    ).toJSON()
    expect(tree).toMatchSnapshot();
})