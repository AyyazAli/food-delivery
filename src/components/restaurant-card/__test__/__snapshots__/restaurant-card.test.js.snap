// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
  data-testid="restaurant-card"
>
  <a
    href="/"
    onClick={[Function]}
    style={
      Object {
        "color": "inherit",
        "textDecoration": "none",
      }
    }
  >
    <button
      className="MuiButtonBase-root MuiCardActionArea-root"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <div
        className="MuiCardHeader-root"
      >
        <div
          className="MuiCardHeader-avatar"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M8.1 13.34l2.83-2.83L3.91 3.5c-1.56 1.56-1.56 4.09 0 5.66l4.19 4.18zm6.78-1.81c1.53.71 3.68.21 5.27-1.38 1.91-1.91 2.28-4.65.81-6.12-1.46-1.46-4.2-1.1-6.12.81-1.59 1.59-2.09 3.74-1.38 5.27L3.7 19.87l1.41 1.41L12 14.41l6.88 6.88 1.41-1.41L13.41 13l1.47-1.47z"
            />
          </svg>
        </div>
        <div
          className="MuiCardHeader-content"
        >
          <h2
            className="MuiTypography-root MuiTypography-h2"
          >
            KFC
          </h2>
        </div>
      </div>
      <div
        className="MuiCardContent-root"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          Best Burgers in town
        </p>
        <p
          className="MuiTypography-root MuiTypography-body2 MuiTypography-colorError"
        >
          Meal Type: 
          fastFood
        </p>
      </div>
      <span
        className="MuiCardActionArea-focusHighlight"
      />
    </button>
  </a>
  <div
    className="MuiCardActions-root MuiCardActions-spacing"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
          />
        </svg>
      </span>
    </button>
  </div>
</div>
`;
